src/System.Text.Json/docs/writable_json_dom_spec.md

```diff
@@ -0,0 +1,240 @@
+    | 2 | - only one additional method <br> - accepts collection of different types <br > | - works  only in C# <br> - no IntelliSense <br> - users may not be aware of it <br> - accepts only Array <br> - accepts collection of not JsonNodes <br> - needs to check it in runtime | from {1,2}, <br>2 seems worse |
+    | 3 | - accepts IEnumerable <br> - does not accept collection of not JsonNodes <br> - no checks in runtime <br> - IntelliSense | - a lot of additional methods <br> - does not accept a collection of different types | gives less possibilities than {1,2}, but requiers no additional checks |
+
+* Implicit operators for `JsonString`, `JsonBoolean` and `JsonNumber` as an additional feature.
+* `Sort` not implemented for `JsonArray`, beacuse there is no right way to compare `JsonObject`s. If user wants to sort `JsonArray` of `JsonNumber`s, `JsonBooleans`s or `JsonStrings` he/she now needs to do the following: convert `JsonArray` to regular array (by iterating through all elements), calling sort (and converting back to `JsonArray` if needed).
+* No support for duplicates of property names. Possibly, adding an option for user to choose from: "first value", "last value", or throw-on-duplicate.
+* Transformation API:
+    * `DeepCopy` method in JsonElement allowing to change JsonElement and JsonDocument into JsonNode recursively transforming all of the elements
+    * `AsJsonElement` method in JsonNode allowing to change JsonNode into JsonElement with IsImmutable property set to false
+    * `IsImmutable` property informing if JsonElement is keeping JsonDocument or JsonNode underneath
+    * `Parse(string)` in JsonNode to be able to parse Json string right into JsonNode if user knows he/she wants mutable version
+    * `DeepCopy` in JsonNode to make a copy of the whole tree
+    * `GetNode` and TryGetNode in JsonNode allowing to retrieve it from JsonElement
+
+## Implementation details
+* `JsonNumber` value is stored as a `string`. 
+
+## Open questions
+API:
+* Do we want to add recursive equals on `JsonArray` and `JsonObject`?
+* Do we want to make `JsonNode`s derived types (and which) implement `IComperable`?
+* Would escaped characters be supported for creating `JsonNumber` from string? 
+
+Implementation:
+* Do we want to add a copy of `JsonWriterHelper.ValidateNumber` with additional checks?
+* Do we want to store `JsonNumber` as `Span<byte>` instead of `string`? 
+
+## Useful links
+
+### JSON
+* grammar: https://www.json.org/
+* specification: https://www.ecma-international.org/publications/files/ECMA-ST/ECMA-404.pdf
+
+### Similar APIs
+`JsonElement` and `JsonDocument` from `System.Json.Text` API:
+* video: https://channel9.msdn.com/Shows/On-NET/Try-the-new-SystemTextJson-APIs
+* blogpost: https://devblogs.microsoft.com/dotnet/try-the-new-system-text-json-apis/
+* performance: https://github.com/dotnet/corefx/issues/33115
```

Comment:
There's a note on the need for high-performance JSON APIs and how it was planned to be achieved. It's something I found useful to read before implementing new API, but I can remove it.

Reply:
I think its good to link to it. I would just update the text to be announcement (since it doesn't really talk much about the actual performance characteristics)

```suggestion
* announcement: https://github.com/dotnet/corefx/issues/33115
```

Reply:
Its perfectly fine to add a link to the announcement. I just thought it didn't make sense to label it as "performance" specifically since there isn't any concrete performance related metrics/measurements in the announcement.

Labeling it announcement makes sense, imo.
```suggestion
* announcement: https://github.com/dotnet/corefx/issues/33115
```